src/components/register/RegisterComponent.js

```diff
@@ -19,121 +124,129 @@ const RegisterComponent = () =>
-                    <input class="form-control" placeholder="Username"/>
+                    <input class="form-control" placeholder="Username"
+                        onChange={(e) => this.updateField("username", e.target.value)
+                        }/>
                 </div>
 
                 <div className="form-group">
                     <label for="password" class="col-form-label text-white float-left">
                         Password
                     </label>
-                    <input type="password" class="form-control" placeholder="Password"/>
+                    <input type="password" class="form-control" placeholder="Password"
+                        onChange={(e) => this.updateField("password", e.target.value)
+                        }/>
                 </div>
 
                 <div className="form-group">
                     <label for="confirm-password" class="col-form-label text-white float-left">
                         Confirm Password
                     </label>
-                    <input type="password" class="form-control" placeholder="Password"/>
+                    <input type="password" class="form-control" placeholder="Confirm Password"
+                        onChange={(e) => this.updateField("confirmPassword", e.target.value)
+                        }/>
                 </div>
 
                 <div class="form-group row">
-                    <div class="form-group col-md-6">
-                        <label for="inputCity" class="col-form-label text-white float-left">City</label>
-                        <input type="text" class="form-control" id="inputCity" placeholder="City"/>
+                    <div class="form-group col-md-4">
+                        <label for="inputContinent" class="col-form-label text-white float-left">Continent</label>
+                        <select id="inputContinent" class="form-control"
+                            onChange={(e) => {
+                                let selectedContinent = e.target.value;
+                                let regionList = this.getRegionsForContinent(selectedContinent);
+                                let selectedRegion = regionList[0];
+                                let subregionList = this.getSubregionsForRegion(selectedContinent, selectedRegion);
+                                let selectedSubregion = subregionList[0][1];
+                    
+                                this.setState(prevState => (
+                                    {
+                                        ...prevState,
+                                        'selectedContinent': selectedContinent,
+                                        'regionList': regionList,
+                                        'selectedRegion': selectedRegion,
+                                        'subregionList': subregionList,
+                                        'selectedSubregion': selectedSubregion
+                                    }))}}>
```

Comment:
i like it! i think this is great as it is right now, but it'd also be nice to have the default values for Continent, Region, and Subregion be `Choose` or `---`, rather than US, Alabama, and Auburn, respectively. This would require a new type of error message for form validation failure when Continent, Region, and Subregion aren't correctly selected though. Time permitting, i'd be happy to go ahead and try to add this.